### PR TITLE
LPS-138683 Fix bug when deleting the target of an ab test

### PR DESCRIPTION
**Pull request by claraizquierdo**
Motivation
=======
[Link to story or ticket](https://issues.liferay.com/browse/LPS-138683)

Proposed Solution
========
When we render SegmentsExperiments component and ClickGoalPicker component we check if the click target exists in the DOM. If it does not exist anymore we reset the Context to target: ''.

Steps to Verify:
----------------
1. Go A/B Test panel > 'Create Test' > Set 'Name' and select 'Click' > 'Save'
2. Select a button as click goal
3. Create a Variant for the test
4. Go to 'Edit' Page > Delete the button element > 'Publish'
5. Go back to the AB test and Click 'Review and Run Test' > 'Run'

The Application should not let you run the test because you have removed the click target from the DOM.

### CODE REVIEW

**Comment by claraizquierdo** [comment]
ci:test:sf

File: modules/dxp/apps/segments/segments-experiment-web/src/main/resources/META-INF/resources/js/components/ClickGoalPicker/ClickGoalPicker.es.js [hunk 1]
```diff
@@ -72,13 +82,12 @@ function ClickGoalPicker({allowEdit = true, onSelectClickGoalTarget, target}) {
...
 	useEffect(() => {
 		ref.current = selectedTarget;
-
 		setSelectorInputValue(selectedTarget);
 	}, [selectedTarget]);
 
 	const previousTarget = ref.current;
 
-	if (selectedTarget != previousTarget) {
+	if (selectedTarget != previousTarget && selectedTarget !== target) {
```
Comment (beltranrengifo) [inline modules/dxp/apps/segments/segments-experiment-web/src/main/resources/META-INF/resources/js/components/ClickGoalPicker/ClickGoalPicker.es.js:90]: Is it possible to use strict equal in both cases?

File: modules/dxp/apps/segments/segments-experiment-web/src/main/resources/META-INF/resources/js/components/ClickGoalPicker/ClickGoalPicker.es.js [hunk 2]
```diff
@@ -62,6 +62,16 @@ function ClickGoalPicker({allowEdit = true, onSelectClickGoalTarget, target}) {
 
 	const {isValidTarget, mode, selectedTarget} = state;
 
+	// If the parent passes as a prop an empty target and the old selected target
+	// is not in the DOM anymore we must update it in the Context
+
+	if (!target && selectedTarget && !document.getElementById(selectedTarget)) {
```
Comment (beltranrengifo) [inline modules/dxp/apps/segments/segments-experiment-web/src/main/resources/META-INF/resources/js/components/ClickGoalPicker/ClickGoalPicker.es.js:68]: Can we condense this condition in a variable with a semantic name? If not the comment is fine for me 👍

File: modules/dxp/apps/segments/segments-experiment-web/src/main/resources/META-INF/resources/js/components/SegmentsExperiments.es.js [hunk 3]
```diff
@@ -68,7 +68,13 @@ function SegmentsExperiments({
 		: selectedExperienceId;
 	const noExperimentIllustration = `${imagesPath}${NO_EXPERIMENT_ILLUSTRATION_FILE_NAME}`;
 	const winnerVariant = variants.find((variant) => variant.winner === true);
-	const goalTarget = experiment?.goal?.target?.substring(1);
+	const goalTarget = experiment?.goal?.target?.replace('#', '');
+
+	// If the target has been removed from the page we must reset it
+
+	if (goalTarget && !document.getElementById(goalTarget)) {
```
Comment (beltranrengifo) [inline modules/dxp/apps/segments/segments-experiment-web/src/main/resources/META-INF/resources/js/components/SegmentsExperiments.es.js:75]: Same as before, let me know what you think => Can we condense this condition in a variable with a semantic name? If not, the comment is fine for me 👍

**Review by beltranrengifo — COMMENTED** [review]
Good job Clara! this bug does not look easy 😅 
Some minor thoughts, just trying to improve readability, if possible...

**Comment by claraizquierdo** [comment]
ci:forward

**Comment by claraizquierdo** [comment]
ci:stop

**Comment by claraizquierdo** [comment]
ci:stop

**Comment by claraizquierdo** [comment]
ci:forward

**Review by beltranrengifo — APPROVED** [review]

**Pull request closed without merging**